src/Manifolds.jl

```diff
@@ -435,68 +417,127 @@ function shortest_geodesic(M::Manifold, x, y, T::AbstractVector)
 end
 
 """
-    vector_transport_to!(M::Manifold, vto, x, v, y)
+    [Ξ,κ] = tangent_orthonormal_basis(M,x,v)
+
+returns both an orthonormal basis `Ξ` for the tangent space of `x` on the
+[`Manifold`](@ref) `M`, which as its first direction contains a normalized
+tangent vector of `v` as well as the sectional curvatures `κ` required for
+computing Jacobi fields, i.e. the first value (along `v`) is zero.
+"""
+tangent_orthonormal_basis(M::Manifold,x,v) = error("A tangent orthonormal basis not implemented on $(typeof(M)) at $(typeof(x)) with first direction $(typeof(v)).")
+
+"""
+    AbstractVectorTransportMethod
+
+An abstract type, which method to use for a vector transport, i.e. within
+[`vector_transport_to`](@ref), [`vector_transport_direction`](@ref) or
+[`vector_transport_along`](@ref). 
+"""
+abstract type AbstractVectorTransportMethod end
+
+"""
+    ParallelTransport <: AbstractVectorTransportMethod
+
+Specify to use parallel transport as vector transport method within
+[`vector_transport_to`](@ref), [`vector_transport_direction`](@ref) or
+[`vector_transport_along`](@ref). 
+"""
+struct ParallelTransport <: AbstractVectorTransportMethod end
+
+"""
+    ProjectTangent <: AbstractVectorTransportMethod
+
+Specify to use projection onto tangent space as vector transport method within
+[`vector_transport_to`](@ref), [`vector_transport_direction`](@ref) or
+[`vector_transport_along`](@ref). See [`project_tangent`](@ref) for details.
+"""
+struct ProjectTangent <: AbstractVectorTransportMethod end
+
+"""
+    vector_transport_to!(M::Manifold, vto, x, v, y, m::AbstractVectorTransportMethod=ParallelTransport())
```

Comment:
I just realized that these vector transport methods implicitly assume that a tangent vector is transported. That's an unrelated issue though.

Reply:
yes, one can easily write the generic version for cotangents using additional flat and sharp.